chat_shell/tests/test_kb_prompt_deduplication.py

```
@@ -0,0 +1,259 @@
+    """Test KB prompt marker detection logic."""
+
+    def setup_method(self):
+        """Set up test fixtures."""
+        # Import here to avoid import issues in non-package tests
+        from chat_shell.interface import ChatRequest
+        from chat_shell.services.context import ChatContext
+
+        self.ChatContext = ChatContext
+        self.ChatRequest = ChatRequest
+
+    def _create_context(self, system_prompt: str = "") -> "ChatContext":
```

Ruff (0.14.13)

Undefined name `ChatContext`

(F821)